### DISCUSSION TITLE
where is data stored?

### DISCUSSION THREAD

**Discussion by fgigon** [SauceLLC/sauce4strava#117] [General]
Hi, I would like to use sauce more and recently became a Patron. Sauce started syncing my activities from Strava, however next time I opened the browser and sauce again it started again at 0 synced activities, maybe because I deleted browser history and cookies in the meantime. Therefore, my question: where does Sauce Activity Sync store the synced activities and is there a possibility to set the location in a way that clearing browser history/cookies wont affect it? Thanks for you help and best regards!

**Comment by mayfield** [comment]
Hi @fgigon 

**How/Why**

Sauce Performance data is stored using IndexedDB (https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) which binds data to a website "origin".  When you manage site settings for a website, what you're doing is manipulating the config or cache for that particular origin.  An origin is defined by the website's protocol, hostname and port (e.g. `scheme://hostname:port`).  In the case of browser extensions like Sauce, the origin is actually something like `chrome-extension://lnoldaojgmennmfncfbmknkkkkdncfid` where that long random number is the identifier for the Sauce extension in particular.

Typically if you just clear site settings for one site, like `https://www.strava.com`, your Sauce Performance data should be unaffected.  In fact it's quite hard to specifically remove the data associated with an extension and nothing else.  You actually need to remove the extension entirely.  On Firefox I think you have to restart after uninstalling as well.  Then reinstall and this will flush out the data.

In your case I assume you cleared ALL browser history (even if just for a period of time) and this will nuke data for ALL origins.  There isn't anything that an extension can do to prevent data loss in this case other than to use remote servers for data storage.  There are number of legal and privacy concerns with doing that, so Sauce is server-less and will remain so.  I truly don't want access to anyone's data.

**Solution**

What I heavily recommend to everyone, is to do the full sync, then make a backup of the data so you never have to go through that again.  The backup is just a single file that you can upload to a dropbox, google drive, etc and then restore from on any devices you have.  It's a good way to use Sauce Performance on multiple machines too, so you just have to run the slow sync on one device once.  After that, the updates are incremental and generally very quick.

Here's a video on the sync mgmt process that covers the aforementioned process:

  https://www.youtube.com/watch?v=k5JAMvsu39A

**Reply by fgigon** [reply]
Hi Justin, many thanks for your extensive and detailed explanations! I will proceed as you recommended, and backup the data after complete sync. Then I can try without risk to find out which browser cleaning action deletes the history. Thanks again and looking forward to finding out more about all the features and possibilities of Sauce. Best regards
